public/app/core/components/RolePicker/UserRolePicker.tsx

```diff
@@ -24,64 +25,41 @@ export const UserRolePicker: FC<Props> = ({
   disabled,
   builtinRolesDisabled,
 }) => {
+  const [roleOptions, setRoleOptions] = useState<Role[]>([]);
+  const [appliedRoles, setAppliedRoles] = useState<Role[]>([]);
+  const [builtInRoles, setBuiltinRoles] = useState<Record<string, Role[]>>({});
+  const [isLoading, setIsLoading] = useState(true);
+
+  useAsync(async () => {
+    try {
+      let options = await (getRoleOptions ? getRoleOptions() : fetchRoleOptions(orgId));
+      setRoleOptions(options.filter((option) => !option.name?.startsWith('managed:')));
+
+      const builtInRoles = await (getBuiltinRoles ? getBuiltinRoles() : fetchBuiltinRoles(orgId));
+      setBuiltinRoles(builtInRoles);
+
+      const userRoles = await fetchUserRoles(userId, orgId);
+      setAppliedRoles(userRoles);
+    } catch (e) {
+      // TODO handle error
+      console.error('Error loading options');
+    } finally {
+      setIsLoading(false);
```

Comment:
nit: I think the is loading parameter is exposed from the `useAsync` so you don't need to handle that state.